oak_sign/src/tests.rs

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use assert_matches::assert_matches;
+use oak_sign::{get_sha256_hex, KeyPair, SignatureBundle};
+
+const TEST_DATA: &str = "Test data";
+const TEST_SHA256: &str = "e27c8214be8b7cf5bccc7c08247e3cb0c1514a48ee1f63197fe4ef3ef51d7e6f";
+
+#[test]
+fn key_pair_test() {
+    let generate_result = KeyPair::generate();
+    assert_matches!(generate_result, Ok(_));
+    let key_pair = generate_result.unwrap();
+
+    let pkcs8_key_pair = key_pair.pkcs8_key_pair();
+    let parse_result = KeyPair::parse(&pkcs8_key_pair);
+    assert_matches!(parse_result, Ok(_));
```

Comment:
This does not check that the resulting key is actually the same as the original one, right?

Reply:
`Eq` is not implemented for `Ed25519KeyPair`

Reply:
But you could re-serialize to pkcs8 and compare that, I guess? or impl Eq for KeyPair, similar to how you did Clone.

Reply:
Done